main.nf

```diff
@@ -2077,6 +2088,58 @@ if ( params.gatk_ug_jar != '' ) {
   """
  }
 
+ // pileupCaller for 'random sampling' genotyping
+
+if (params.pileupcaller_bedfile.isEmpty()) {
+  ch_bed_for_pileupcaller = file('NO_FILE_BED')
+} else {
+  ch_bed_for_pileupcaller = Channel.fromPath(params.pileupcaller_bedfile)
+}
+
+if (params.pileupcaller_snpfile.isEmpty ()) {
+  ch_snp_for_pileupcaller = file('NO_FILE')
+} else {
+  ch_snp_for_pileupcaller = Channel.fromPath(params.pileupcaller_snpfile)
+}
+
+
+
+ process genotyping_pileupcaller {
+  label 'mc_small'
+  tag "${samplename}"
+  publishDir "${params.outdir}/genotyping", mode: 'copy'
+
+  when:
+  params.run_genotyping && params.genotyping_tool == 'pileupcaller'
+
+  input:
+  tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, file(bam), file(bai) from ch_damagemanipulation_for_genotyping_pileupcaller
+  file fasta from ch_fasta_for_genotyping_pileupcaller.collect()
+  file fai from ch_fai_for_pileupcaller.collect()
+  file dict from ch_dict_for_pileupcaller.collect()
+  file bed from ch_bed_for_pileupcaller
+  file snp from ch_snp_for_pileupcaller
+
+  output:
+  tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, file("pileupcaller.*")
+
+  script:
+  caller = "--${params.pileupcaller_caller}"
+  if (params.pileupcaller_majority_call) {
```

Comment:
If this a binary decision you can simplify this do the: `params.pileupcaller_majority_call ? majority : randomdiplooid`

But if you can run the command without either of those two options, you should add an extra check in the sanity checking at the top.

Reply:
I see from `nextflow.config` you hve two flags, one which you supply a string for `randomDiploid` and another where it's a boolean flag? What's the difference there?